Apps/Common/SIMSolver.h

```diff
@@ -123,10 +123,14 @@ template<class T1> class SIMSolver : public SIMSolverStat<T1>
   explicit SIMSolver(T1& s1) : SIMSolverStat<T1>(s1,"Time integration driver")
   {
     saveDivergedSol = false;
+    restartAdm = nullptr;
   }
 
-  //! \brief Empty destructor.
-  virtual ~SIMSolver() {}
+  //! \brief The destructor closes the restart data handler.
+  virtual ~SIMSolver()
+  {
+    delete restartAdm;
+  }
```

Comment:
the destructor can still be on a single line. And in the doxy, maybe `deletes` is more correct that `closes` ?